ext/sockets/sockets.c

```diff
@@ -1217,6 +1217,11 @@ PHP_FUNCTION(socket_write)
 		length = str_len;
 	}
 
+	if (length < 0) {
+		php_error_docref(NULL, E_WARNING, "Length cannot be negative");
+		RETURN_FALSE;
+	}
```

Comment:
Why not move this if statement up, so that immediately follows the `zend_parse_parameter()` call?

Reply:
Yeah, I was thinking about the previous block that changes `length` but now I see that str_len would never be negative. Will move it up

Reply:
Will move it up. as KalleZ mentioned I was thinking of the block above. But str_len would never be negative.